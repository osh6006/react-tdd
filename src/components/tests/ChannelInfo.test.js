import { Route } from "react-router-dom";
import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";

import ChannelInfo from "../ChannelInfo";

import { withAllContexts, withRouter } from "../../tests/utils";

describe("channel Info", () => {
  const fakeYoutube = {
    channelImageURL: jest.fn(),
  };

  afterEach(() => fakeYoutube.channelImageURL.mockReset());

  it("renders correctly", async () => {
    const { asFragment } = renderChannelInfoWithCallback(() => "url");

    await waitFor(() => screen.findByText("channel"));
    await waitFor(() => screen.findByRole("img"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without img correctly", async () => {
    renderChannelInfoWithCallback(() => {
      throw new Error("error");
    });

    expect(screen.queryByRole("img")).toBeNull();
  });

  function renderChannelInfoWithCallback(callback) {
    fakeYoutube.channelImageURL.mockImplementation(callback);
    return render(
      withAllContexts(
        withRouter(
          <Route path="/" element={<ChannelInfo id="id" name="channel" />} />
        ),
        fakeYoutube
      )
    );
  }
});
