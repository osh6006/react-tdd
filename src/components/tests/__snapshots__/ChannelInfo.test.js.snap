// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channel Info renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex my-4 mb-8 items-center"
  >
    <img
      alt="channel"
      class="w-10 h-10 rounded-full"
      src="url"
    />
    <p
      class="text-lg font-medium ml-2"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;

exports[`channel Info renders without correctly 1`] = `
<DocumentFragment>
  <div
    class="flex my-4 mb-8 items-center"
  >
    <p
      class="text-lg font-medium ml-2"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;
